moveit_core/dynamics_solver/src/dynamics_solver.cpp

```diff
@@ -281,7 +281,7 @@ bool DynamicsSolver::getMaxPayload(const std::vector<double>& joint_angles, doub
 }
 
 bool DynamicsSolver::getPayloadTorques(const std::vector<double>& joint_angles, double payload,
-                                       std::vector<double>& joint_torques) const
+                                       std::vector<double>& joint_torques) const // NonPure, and again they depend on the joint_model_group_ boolean variable
```

Comment:
I think just one warning in the header file would be enough, please. Labeling every singe function is excessive IMO.

Can you please delete all comments here and put a warning like this in `dynamics_solver.h`?

// Warning: many of these functions are "impure", meaning their output depends on the state of member variables in a non-transparent way. For example, `getMaxPayload()` depends on `joint_model_group_`.

Reply:
OK, sir, I'll work on modifying them today.